### ISSUE TITLE
Support variable refresh rate monitors

### ISSUE THREAD

**Issue by leocb** [clsid2/mpc-hc#555]
TL;DR
Add an option to draw frames at the same rate as the content's fps and disable v-sync when using D3D fullscreen, this way the VRR will kick in.

I don't know if this is already done in any extent, but I had an idea
Since we have variable refresh rate monitors (G-Sync or Freesync), my idea is to add an option to use an exclusive fullscreen when drawing frames **and only update the screen at the same rate as the content fps**.

If I understand correctly, currently it draws the frame as fast as possible without tearing - conventional v-sync - matching the maximum refresh rate of the monitor, in my case 144hz: I can verify this by enabling the g-sync frame counter on my monitor.
Drawing at the same rate as the content fps will make the GPU match the content's refresh rate on the monitor, giving us for example, a true [judder-free](https://www.rtings.com/tv/tests/motion/24p) 24fps experience

Note 1: At least on G-sync, there's no need to worry about duplicating frames for low frame-rates, as the g-sync module will do it (as seen [on this blur busters post on item 4](https://blurbusters.com/gsync/gsync101-input-lag-tests-and-settings/2/))

Note 2: 24hz Judder can only be observed on 60hz and 180hz monitors as all other refresh rates can be divided by 24 without a remainder. However, 144hz cannot be divided evenly for 30 or 60fps content. My point is, even if some specific content will play well for your refresh rate, some others may not, hence why I believe VRR will be a good fit for this project

Here's some comparison shots of the judder on my 144hz monitor, all tests were made using VMR-9 (Renderless) with D3D Fullscreen

To add to the injury, it seems like the current v-sync solution is not that stable since judder appears on irregular intervals as noticeable on the 60fps test.

24fps is stable, as 144/24 = 6 (same exposure on all squares)
![24hz](https://user-images.githubusercontent.com/8310271/88756696-ad0f7080-d13a-11ea-8747-90e71015248b.jpg)

30fps judder 144/30 = 4.8 (uneven exposure on squares indicates judder)
![30hz](https://user-images.githubusercontent.com/8310271/88756691-ab45ad00-d13a-11ea-8516-4df85273031d.jpg)

60fps judder 144/60 = 2.4 (uneven exposure on squares indicates judder)
![60hz](https://user-images.githubusercontent.com/8310271/88756695-ac76da00-d13a-11ea-9c9a-56e8c14dc5ee.jpg)

**Comment by clsid2** [comment]
No new functionality will be added to the internal renderers.

You will have to request this in an external renderer such as madVR. But while VRR might sound useful for video in theory, it is far from trivial to implement.
https://forum.doom9.org/showpost.php?p=1717019&postcount=28759

**Comment by leocb** [comment]
Understood, thanks!
